torch/csrc/dynamo/eval_frame.c

```diff
@@ -137,6 +137,10 @@ THPPyInterpreterFrame* THPPyInterpreterFrame_New(_PyInterpreterFrame* frame) {
   } else {                                                              \
   }
 
+// Uncomment next line to print DEBUG_TRACE messages
```

Comment:
The changes here are really extensive now, but there is zero PR description, can we get at least some high level description of the strategy please?